File: 2022/go/day01/day01_test.go
package day01_test

/*
rg --files | entr -cr go test aoc/day01
*/

import (
	"aoc/day01"
	"reflect"
	"testing"
)

var INPUT = `
1000
2000
3000

4000

5000
6000

7000
8000
9000

10000
`

func TestToCalories(t *testing.T) {
	got := day01.ToCalories(INPUT)
	expected := [][]int{{1000, 2000, 3000}, {4000}, {5000, 6000}, {7000, 8000, 9000}, {10000}}
	if !reflect.DeepEqual(got, expected) {
		t.Errorf("Got: %v; want: %v", got, expected)
	}
}

func TestToCaloriesAgg(t *testing.T) {
	got := day01.ToCaloriesAgg(INPUT)
	expected := []int{6000, 4000, 11000, 24000, 10000}
	if !reflect.DeepEqual(got, expected) {
		t.Errorf("Got: %v; want: %v", got, expected)
	}
}

func TestMostCalories(t *testing.T) {
	want := 24000
	got := day01.MostCalories(INPUT)
	if got != want {
		t.Errorf("Want: %d, got: %d", want, got)
	}
}

func TestTop3Calories(t *testing.T) {
	want := 45000
	got := day01.Top3Calories(INPUT)
	if got != want {
		t.Errorf("Want: %d, got: %d", want, got)
	}
}

File: 2022/ruby/test/day01_test.rb
# frozen_string_literal: true

require 'minitest'
require 'minitest/autorun'
require 'lib/day01'

# rg --files lib/ test/ | entr -cr ruby -I .:test ./test/day01_test.rb
class Day01Test < Minitest::Test

  INPUT = <<~INPUT
    1000
    2000
    3000

    4000

    5000
    6000

    7000
    8000
    9000

    10000
  INPUT

  def setup
    @day = Day01.new(INPUT)
  end

  def test_calories
    expected = [[1000, 2000, 3000], [4000], [5000, 6000], [7000, 8000, 9000], [10_000]]
    assert_equal expected, @day.calories
  end

  def test_most_calories
    assert_equal 24_000, @day.most_calories
  end

  def test_top3_calories
    assert_equal 45_000, @day.top3_calories
  end
end
